firefox_ui_tests/functional/locationbar/test_star_in_autocomplete.py

```diff
@@ -45,15 +49,18 @@ def test_star_in_autocomplete(self):
         with self.marionette.using_context('content'):
             self.marionette.navigate(self.marionette.absolute_url('layout/mozilla_grants.html'))
 
+        # TODO: Convert to l10n friendly accessor when menu library is available
         self.browser.menubar.select('Bookmarks', 'Bookmark This Page')
+        # TODO: Replace hard-coded selector with library method when one is available
         done_button = self.marionette.find_element('id', 'editBookmarkPanelDoneButton')
+        self.wait_for_condition(lambda mn: done_button.is_displayed)
         done_button.click()
 
-        self.clear_history()
+        self.clean_up_state()
 
         locationbar = self.browser.navbar.locationbar
-        # TODO: Replace with visited observer when places module is available
-        # https://bugzilla.mozilla.org/show_bug.cgi?id=1121731
+        # TODO: Replace with places.wait_for_visited when observers have been implemented
+        # https://bugzilla.mozilla.org/show_bug.cgi?id=1121691
         # The data for autocomplete does not load immediately and we cannot wait for an observer
         # until the bug 1121731 is complete
         time.sleep(4)
```

Comment:
Oh, you want to make sure this is called when you navigate to the test page. So use marionette.navigate() inside the callback.